pyphare/pyphare_tests/test_pharesee/test_geometry_2d.py

```diff
@@ -0,0 +1,493 @@
+                        Box([40, 27], [40, 40]),
+                        Box([28, 27], [39, 27]),
+                        Box([28, 40], [39, 40]),
+                    ]
+                },
+            ],
+        }
+
+        gaboxes = ghost_area_boxes(hierarchy, "particles")
+        particles = "particles"
+
+        self.assertEqual(len(expected), len(gaboxes))
+
+        for ilvl, lvl in enumerate(hierarchy.patch_levels):
+            for act_pdata, exp_pdata in zip(gaboxes[ilvl][particles], expected[ilvl]):
+                self.assertEqual(len(exp_pdata["boxes"]), len(act_pdata["boxes"]))
+                for exp_box in exp_pdata["boxes"]:
+                    self.assertTrue(exp_box in act_pdata["boxes"])
+
+
+
+    @data(
+        (
+            {
+                1: [
+                    {"boxes": [Box([9, 9], [9, 20])]},
+                    {"boxes": [Box([20, 9], [20, 20])]},
+                    {"boxes": [Box([10, 9], [19, 9])]},  # verify
+                    {"boxes": [Box([10, 20], [19, 20])]},  # verify
+                    {"boxes": [Box([27, 27], [27, 40])]},
+                    {"boxes": [Box([40, 27], [40, 40])]},
+                    {"boxes": [Box([28, 27], [39, 27])]},  # verify
+                    {"boxes": [Box([28, 40], [39, 40])]},  # verify
+                ]
+            },
+        ),
+    )
+    @unpack
+    def test_particle_level_ghost_boxes(self, expected):
+        print("GeometryTest.test_particle_level_ghost_boxes")
+        hierarchy = super().basic_hierarchy()
+
+        lvl_gaboxes = level_ghost_boxes(hierarchy, "particles")
+        for ilvl in range(1, len(hierarchy.patch_levels)):
+            self.assertEqual(len(lvl_gaboxes[ilvl].keys()), 1)
+
+            key = list(lvl_gaboxes[ilvl].keys())[0]
+
+            self.assertEqual(len(expected[ilvl]), len(lvl_gaboxes[ilvl][key]))
+
+            for actual, exp in zip(lvl_gaboxes[ilvl][key], expected[ilvl]):
+
+                for act_box, exp_box in zip(actual["boxes"], exp["boxes"]):
+                    self.assertEqual(act_box, exp_box)
+
+
+
+    def test_particle_level_ghost_boxes_do_not_overlap_patch_interiors(self):
+        print("GeometryTest.test_particle_level_ghost_boxes_do_not_overlap_patch_interiors")
+        hierarchy = super().basic_hierarchy()
+
+        lvl_gboxes = level_ghost_boxes(hierarchy, "particles")
+
+        assert len(lvl_gboxes) > 0
+        for ilvl, pdatainfos in lvl_gboxes.items():
+            assert len(pdatainfos) > 0
+            for particles_id, gaboxes_list in pdatainfos.items():
+                assert len(gaboxes_list) > 0
+                for pdatainfo in gaboxes_list:
+                    for box in pdatainfo["boxes"]:
+                        for patch in hierarchy.patch_levels[ilvl].patches:
+                            self.assertIsNone(patch.box * box)
+
+
+
+@ddt
+class ParticleLevelGhostGeometryTest(AGeometryTest):
+    @data(
+        (
+            {"L0": [Box2D(5, 9), Box2D(14, 19)]},
```

Comment:
I don't understand how that test is different from the  AGeometryTest::test_particle_level_ghost_boxes the forst test case seems exactly the same ?

Reply:
mostly the same, there's multiple configurations in this one, so I can remove the other one which I guess was a copy/paste from the 1d version